### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by myaserkhan**
**For the second milestone of the OPP School Library project covered the following requirements**

- Create a class Nameable. with a method called correct_name that will raise a NotImplementedError.
- Make Person class inherits from Nameable and add a method correct_name which returns the name attribute.
- Create BaseDecorator class that inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable and Implement the correct_name method that returns the result of the correct_name method of the @nameable.
- Create CapitalizeDecorator class that inherits from BaseDecorator and Implement a method correct_name that capitalizes the output of @nameable.correct_name.
- Create TrimmerDecorator class that inherits from BaseDecorator and Implement a method correct_name that makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: Decorator/trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require_relative './decorator'
```
Comment (TedLivist) [inline Decorator/trimmer_decorator.rb]: - [x] Kindly fix your class import here. It should be:

  ```
  require_relative './base_decorator'
  ```

File: Entities/person.rb [hunk 2]
```diff
@@ -1,8 +1,11 @@
-class Person
+require_relative '../Decorate/nameable'
```
Comment (TedLivist) [inline Entities/person.rb]: - [ ] Kindly fix your file import here.

  ```
  require_relative '../Decorator/nameable'
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @myaserkhan,

Good job so far!
✔️ Implementation of class inheritance
✔️ Good documentation
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @myaserkhan👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**